drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -597,20 +597,20 @@ func (c *portworxBasic) refreshTokenSecret(secret *v1.Secret, cluster *corev1.St
 	if err != nil {
 		return err
 	}
-	secret.Data[PxSaTokenRefreshTimeKey] = []byte(time.Now().UTC().Add(time.Duration(expirationSeconds/2) * time.Second).Format(time.RFC3339))
 	newToken, err := generatePxSaToken(cluster, expirationSeconds)
 	if err != nil {
 		return err
 	}
-	secret.Data[core.ServiceAccountTokenKey] = newToken
+	secret.Data[core.ServiceAccountTokenKey] = []byte(newToken.Status.Token)
+	secret.Data[PxSaTokenRefreshTimeKey] = []byte(time.Now().UTC().Add(time.Duration(*newToken.Spec.ExpirationSeconds/2) * time.Second).Format(time.RFC3339))
```

Comment:
Let's add a comment here to indicate why we are doing this. It is a bit subtle.

Reply:
Agree